nested navigator

SoleCincis opened the pull request:
Current state of the app : 

https://user-images.githubusercontent.com/43641321/138263751-9d66665a-c465-4234-8a43-8813912280b6.MP4

SoleCincis committed:
chore: git rebase due SDK upgrade

SoleCincis committed:
chore: rebase due SDK upgrade

SoleCincis committed:
feat(button): implemented button component

-first steps plus and minus logics

SoleCincis committed:
feat(style): positioned elements on colorGenerator screen

SoleCincis committed:
feat(plus&minus 10 logic): use state to update the value

- using setState to update the value
 - but each value gets updated simultaneously and that's not what I want
- next step is to create a state for each color ?
- value box is not currently in use , cause of state in children component issue

SoleCincis committed:
feat(ColorGeneratorScreen): red, green, blu state

- all logic for plus and minus it's happening in this component at the moment
- there is a lot of code repetition tree different state and six different setState functions
- next step will be a refactor of the all component

SoleCincis committed:
feat(random value): implement a random value logic

- now random value button generates the same random value for all color values
- refactor is necessary
 - three separate fils red, green and blu value component

SoleCincis committed:
feat: implemented random value to change color background

- now if press on New color button background gets updated with latest color values in state
- in button component trying to implement a scale animation

SoleCincis committed:
chore(colorGeneratoScreen): refactore of the component

SoleCincis committed:
test

SoleCincis committed:
feat(nested navigators): create newColorScreen

- in app.js new stack screen not belonging to tabNavigation
- in ColorGenerator new button uses navigate method to go to the new color nested screen
 - it should bring backgroundColor params with it , but probably this is not the correct way to do it with nested navigators
NewColorScreen is not receiving param which is still undefined

SoleCincis committed:
feat(NewColorScreen): passing the right param

-previously was passing the all object newBackgroundColo to params
 - the correct way is to pass just key value of the object backgroundColor
- in this component we are rendering again ColorGeneratorScreen component wrapped in a view with generated backgroundColor coming in as a prop

SoleCincis committed:
feat(ColorGeneratorScreen): component refactor

- plus/minus 10 logics are now using the sexy method Math.min/max
 - which returns the smaller/larger of a set of supplied numeric expressions
- the is no need for GenerateColor state anymore
 - rgbColor is now used as prop and passed to the nested screen

SoleCincis committed:
feat(ColorGeneratScree): implemented randomValue function

- it's out of my component because it dosen't depend on any data from it
- it's applied to use state so when screen is rendered for the first time there is always a fresh rand value

SoleCincis commented on screens/ColorGeneratorScreen.js:
```diff
@@ -1,45 +1,42 @@
     return (
-
-    <View style={[styles.container, { backgroundColor: GeneratedColor } ]}>
+      <View style={[styles.container, { backgroundColor: props }]}>
```
Sorry. Somehow my answer to your question here didn't joined my review. 
Here it's decided that this View has some style, stored in `.container` AND there is a backgroundColor its value though is not yet specified, instead there are props used as a "place holder" which allowed me to declare backgroundColor's value later on .

SoleCincis replied:
Well finally solved in 89303d53080a744cae51b84ab37dda0ffed85d80

-  answer to your second question : Props are objects used to pass data between components, of which data flow is Uni-directional ( from Parent to child to grandchild ) and the only action you can do with the data you get from props is read it . How decent is this explanation ? 
- to answer your first question : it does absolutely nothing. There for it has been removed in my last commit. Here a summery of my process to get to this conclusion : 
after consol.logging what I thought is the  backgroundColor prop that I'm passing to the child component 
<img width="390" alt="Screenshot 2021-11-03 at 16 24 42" src="https://user-images.githubusercontent.com/43641321/140128151-fcb4ea4f-ed64-4a3a-a2db-7486fc313834.png">
I looked at this out put getting this confirmation: I cannot be passing anything to the child because in my 1st log backgroundColor has an undefined value but then in my 2nd log inside the child backgroundColor has the correct value that I wanted , how is that possible ? well because I'm using params to pass data to the child there fore props are unnecessary here.

digorath commented on screens/NewColorScreen.js:
```diff
@@ -0,0 +1,19 @@
+
+  return (
+    <View style={[styles.container, { backgroundColor: backgroundColor} ]}>
+      <ColorGeneratorScreen navigation={navigation}  />
```
this goes back to the other conversation. please try to figure out, what exactly happens currently.

SoleCincis replied:
so now this is clear . Here I'm just using my param backgroundColor assign to backgroundColor css tag(?) for this new component which happens to render its parent with the only change of the background color, that's why I need to use a view around the parent in order to change its background color.

digorath replied:
`this new component which happens to render its parent with the only change of the background color` i do not quite understand what you want to say, sorry

SoleCincis replied:
Let me refries : 
NewColorScreen component is rendering its parent component with a different backgroundColor, therefor it uses a view to wrap its parent in order to apply a different backgroundColor on it.

SoleCincis committed:
chore+feat: rebased on master due sdk upgrade + deleted unused props

- passing the generated rgb color trough parameters not trough props which in this case are
  - not just used incorectely but also not necessary
- changes on yarn and package.json are coming from SDK  upgrade
